aldryn_people/tests/frontend/integration/specs/spec.people.crud.js

```diff
@@ -4,16 +4,196 @@
+
+                        // validate/click edit page link
+                        peoplePage.editPageLink.click();
+
+                        // switch to default page content
+                        browser.switchTo().defaultContent();
+
+                        browser.wait(function () {
+                            return browser.isElementPresent(peoplePage.testLink);
+                        }, peoplePage.mainElementsWaitTime);
+
+                        // validate test link text
+                        peoplePage.testLink.getText().then(function (title) {
+                            expect(title).toEqual('Test');
+                        });
+                    }
+                });
+            }
+        });
+    });
+
+    it('creates a new group', function () {
+        // check if the focus is on sidebar ifarme
+        peoplePage.editPageLink.isPresent().then(function (present) {
+            if (present === false) {
+                // wait for modal iframe to appear
+                browser.wait(function () {
+                    return browser.isElementPresent(peoplePage.sideMenuIframe);
+                }, peoplePage.iframeWaitTime);
+
+                // switch to sidebar menu iframe
+                return browser.switchTo().frame(browser.findElement(By.css(
+                    '.cms_sideframe-frame iframe')));
+            }
+        }).then(function () {
+            browser.wait(function () {
+                return browser.isElementPresent(peoplePage.breadcrumbsLinks.first());
+            }, peoplePage.mainElementsWaitTime);
+
+            // click the Home link in breadcrumbs
+            peoplePage.breadcrumbsLinks.first().click();
+
+            browser.wait(function () {
+                return browser.isElementPresent(peoplePage.groupsLinks.first());
+            }, peoplePage.mainElementsWaitTime);
+
+            peoplePage.groupsLinks.first().click();
+
+            // wait for iframe side menu to reload
+            browser.wait(function () {
+                return browser.isElementPresent(peoplePage.addConfigsButton);
+            }, peoplePage.mainElementsWaitTime);
+
+            // check if the group already exists and return the status
+            return peoplePage.editConfigsLink.isPresent();
+        }).then(function (present) {
+            if (present === false) {
+                // group is absent - create new group
+                browser.wait(function () {
+                    return browser.isElementPresent(peoplePage.addConfigsButton);
+                }, peoplePage.mainElementsWaitTime);
+
+                peoplePage.addConfigsButton.click();
+
+                browser.wait(function () {
+                    return browser.isElementPresent(peoplePage.languageTabs.get(1));
+                }, peoplePage.mainElementsWaitTime);
+
+                // switch to English language tab
+                peoplePage.languageTabs.get(1).click().then(function () {
```

Comment:
@mkoistinen I already tested possible refactoring - it can't be done as we can't break ```if``` condition with not nested ```.then```. BTW ```.then``` in Protractor doesn't work exactly the same as in ES6